source/core/NumericString.cpp

```diff
@@ -1377,9 +1412,12 @@ Boolean TypedScanString(SubString* inputString, IntIndex* endToken, const Format
     
     switch (argumentType->BitEncoding()) {
         case kEncoding_UInt: 
-        case kEncoding_Enum:
             IntScanString(argument, argumentType, formatOptions->FormatChar, inpBegin, &endPointer);
             break;
+        case kEncoding_Enum:
+            if (!EnumScanString(&in, argument, argumentType, formatOptions->FormatChar, inpBegin, &endPointer))
```

Comment:
I wonder if EnumScanString() implementation should handle only the "if case". And call EnumScanString() if (formatChar == 's') and else call IntScanString() from here at this call site. I am not sure whether that would be the right thing to do?